sympy/core/mod.py

```diff
@@ -107,6 +107,16 @@ def doit(p, q):
             elif (qinner*(q + qinner)).is_nonpositive:
                 # |qinner| < |q| and have different sign
                 return p
+        elif isinstance(p, Add):
+            #separating into modulus and non modulus
+            both_l = non_mod_l, mod_l = [], []
```

Comment:
🥇 OK, this is a cool approach to binary sifting! I've copied it to a couple of places in #13599. Is this your own or did you learn this from somewhere else?

Reply:
@smichr  thanks, I learnt it from a comment on a query in StackOverflow.